### PR TITLE
Updated audiobook sleep timer stored value so the can be resumed

### PR DESCRIPTION
**Pull request by nunommts**
**What's this do?**
This PR updates the existing logic to save/restore the current audio book sleep timer value but it now updates the value every time the timer changes so the remaining duration can be stored and the timer doesn't get restarted every time the audiobook is opened.

**Why are we doing this? (w/ JIRA link if applicable)**
There's a comment in [this ticket](https://www.notion.so/lyrasis/Audiobook-sleep-timer-not-saved-after-closing-book-bab815834a374893a0588ca96fc8caa7) saying the timer shouldn't be restarted when reopening the audiobook but resumed.

**How should this be tested? / Do these changes have associated tests?**
Follow [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/55) testing steps once that and [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/56) are merged

**Dependencies for merging? Releasing to production?**
[This PR](https://github.com/ThePalaceProject/android-audiobook/pull/55) and [this PR](https://github.com/ThePalaceProject/android-audiobook/pull/56) need to be merged first

**Have you updated the changelog?**
Yes

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Tested by @nunommts

### CODE REVIEW

**Review by ray-lee — COMMENTED** [review]
@nunommts For some reason the checks didn't run on this PR. Could you push a commit, to get them to run? There's a conflict in the changelog that needs to be resolved anyway.

Also, does this work when the "End of file" sleep timer is selected? I'm curious, since that isn't a specific amount of time.

**Comment by nunommts** [comment]
@ray-lee Actually, that wouldn't work because there's more than a way to handle the "null" values. I'm updating it right now, but I'll need to create a new PR for the audiobook submodule that needs to be merged first.

**Review by ray-lee — COMMENTED** [review]
@nunommts The checks are failing because there's a lint error.

**Comment by nunommts** [comment]
@ray-lee Fixed!

File: simplified-profiles/src/main/java/org/nypl/simplified/profiles/ProfileDescriptionJSON.kt [hunk 1]
```diff
@@ -408,7 +407,11 @@ object ProfileDescriptionJSON {
     )
 
     return map.mapValues { entry ->
-      PlayerSleepTimerConfiguration.valueOf(entry.value)
+      try {
+        entry.value.toLong()
+      } catch (exception: NumberFormatException) {
+        null
```
Comment (ray-lee) [inline simplified-profiles/src/main/java/org/nypl/simplified/profiles/ProfileDescriptionJSON.kt:413]: Does this mean that if there's an unexpected/non-numeric value in the json, the sleep timer will be set to end-of-file? I think it should be set to be off if there is an unexpected value. You may need to check for null explicitly before calling `toLong`, in order to check for end-of-file.

Also, from the test, it looks like older versions of the app were saving values like "MINUTES_15" and "MINUTES_30", right? If so, I think we need to check for these as well, for backward compatibility.

Comment (ray-lee) [reply simplified-profiles/src/main/java/org/nypl/simplified/profiles/ProfileDescriptionJSON.kt:417]: @nunommts Yes, you're right. If restoring the sleep timers wasn't working before, then I don't think we need to worry about handling those old values.

**Review by ray-lee — APPROVED** [review]
Looks good!

**Pull request merged**